src/lumigo_tracer/lumigo_utils.py

```diff
@@ -547,21 +549,38 @@ def _recursive_omitting(
     d, free_space = prev_result
     if free_space < 0:
         return d, free_space
+    should_skip_key = False
+    if isinstance(d, list):
+        d.append(None)
+        key = -1
+    elif omit_skip_path:
+        should_skip_key = omit_skip_path[0] == key
+        omit_skip_path = omit_skip_path[1:] if len(omit_skip_path) > 1 and should_skip_key else None
     if key in SKIP_SCRUBBING_KEYS:
         d[key] = value
         free_space -= len(value) if isinstance(value, str) else len(aws_dump(d))
-    elif isinstance(key, str) and regex and regex.match(key):
+    elif isinstance(key, str) and regex and regex.match(key) and not should_skip_key:
         d[key] = "****"
         free_space -= 4
     elif isinstance(value, (dict, OrderedDict)):
         d[key], free_space = reduce(
-            lambda p, i: _recursive_omitting(p, i, regex, enforce_jsonify),
+            lambda p, i: _recursive_omitting(
+                p, i, regex, enforce_jsonify, omit_skip_path=omit_skip_path
+            ),
             value.items(),
             ({}, free_space),
         )
     elif isinstance(value, decimal.Decimal):
         d[key] = float(value)
         free_space -= 5
+    elif isinstance(value, list):
+        d[key], free_space = reduce(
+            lambda p, i: _recursive_omitting(
+                p, (-1, i), regex, enforce_jsonify, omit_skip_path=omit_skip_path
+            ),
+            value,
+            ([], free_space),
+        )
```

Comment:
I don't really understand what you did here. it looks like you tried to build a new logic on the existing function, although it just makes it more complex (relevant to also lines 553-555).
What's the idea of appending None at the end? what is the meaning of key=-1?
Here's my idea:
This function should never get a list as it's input. 
Instead, you need to create another function, "list_recursive_omitting", that will handle lists. Then, these two functions should call one another when it's relevant.
The code will be much more clean and easy to understand IMO